storage/replica.go

```diff
@@ -202,7 +203,7 @@ type Replica struct {
 	// Last index applied to the state machine. Updated atomically.
 	appliedIndex uint64
 	configHashes map[int][]byte // Config map sha256 hashes @ last gossip
-	systemDBHash []byte         // sha256 hash of the system config @ last gossip
+	systemDBHash []byte         // sha1 hash of the system config @ last gossip
```

Comment:
I doubt this is a decision we need to make based on performance - how much time would we actually spend doing this computation? If we do this so often that performance matters a factor of 2 won't make much difference. I vote to stick with sha256, although I wouldn't object too much to switching to sha1. I don't think we should go to anything weaker than that.

Hash performance is a funny thing. MD5 used to be a lot faster than SHA1, but now it's actually slower on modern servers because intel built specialized sha1 and sha256 instructions into their recent CPUs (I don't know if the go functions currently take advantage of these, but they could. SHA1 is still faster than SHA256, at least until intel decides that SHA1 is obsolete enough that they can remove the special instructions). Over time hashes that are "current" will get faster while older ones get (relatively) slower.